Sources/XcodeGraph/Models/BuildAction.swift

```diff
@@ -2,22 +2,70 @@ import Foundation
 import Path
 
 public struct BuildAction: Equatable, Codable, Sendable {
+    /// It represents the reference to a target from a build action, along with the actions when the target
+    /// should be built.
+    public struct Target: Equatable, Codable, Sendable {
+        /// Xcode project actions when a build scheme action can build a target.
+        public enum BuildFor: Codable, CaseIterable, Sendable {
+            case running, testing, profiling, archiving, analyzing
+        }
+
+        /// The target reference.
+        public var reference: TargetReference
+
+        /// A list of Xcode actions when a target should build.
+        public var buildFor: [BuildFor]?
+
+        public init(_ reference: TargetReference, buildFor: [BuildFor]?) {
+            self.reference = reference
+            self.buildFor = buildFor
+        }
+    }
+
     // MARK: - Attributes
 
-    public var targets: [TargetReference]
+    public var targetsWithBuildFor: [Target]
```

Comment:
I couldn't come up with a better name for this or a different approach to make it backward compatible. Any suggestion for improvements?